resources/lib/player/players.py

```diff
@@ -144,18 +144,28 @@ def _set_external_ids(self, tmdb_type, tmdb_id, season, episode, details, requir
             details.set_details(details=self.details_ext_ids, reverse=True)
         return set_detailed_item(tmdb_type, tmdb_id, season, episode, details=details) or {}
 
-    def _set_provider_priority(self):
-        try:  # Check if players are listed in providers and bump priority if found
-            providers = self.details.infoproperties['providers']
-            providers = providers.split(' / ')
-            for k, v in self.players.items():
-                if 'provider' not in v or v['provider'] not in providers:
-                    v['priority'] = v.get('priority', PLAYERS_PRIORITY) + 100  # Increase priority baseline by 100 to prevent other players displaying above providers
-                    continue
-                v['priority'] = providers.index(v['provider']) + 1  # Add 1 because sorted() puts 0 index last
-                v['is_provider'] = True
-        except (KeyError, AttributeError):
-            pass
+    def _get_prioritised_players(self):
+        try:
+            providers = self.details.infoproperties.get('providers', '')
+        except AttributeError:
+            providers = ''
+        providers = tuple(providers.split(' / '))
```

Comment:
What's the advantage of converting the list from .split into a tuple?

Also note that `''.split(' / ')` will give `['']` which potentially will cause false positive matches - e.g. if a player file has `provider: ""` rather than not including it. See my comment below for a simple workaround.

Reply:
Computer says no.

I use pylint and it complains about using a mutable default value for the `_providers` parameter of `_set_priority`.

A tuple being equivalent to an immutable list makes it happy, but it doesn't bother me either way, especially as there is no modification happening.

Reply:
Ah I see. Yeah dict as default arg is bad.

More to the point though: there's not really any need for the `_providers=` arg at all.

`_set_priority()` is an inner function of `_get_prioritised_players` so it has access to its variable scope. You may as well skip defining `_providers` entirely and just use `providers` directly instead -- unless there's something I missing here?

Reply:
No, not missing anything, it's just a (bad) habit I developed trying to speed up nested or looped function calls.

In theory it makes the variable lookup faster, without having to do the extra dereference, but in practice only makes a difference in much longer loops.

[250c735](https://github.com/jurialmunkey/plugin.video.themoviedb.helper/pull/965/commits/250c7353c1e4438afdc9fb6c03097b7890d7f7a5) cleans things up a bit, and also avoids the list membership check if either `providers` or `player_provider` is not defined or an empty string.

Reply:
Ah I see. Yeah might be a bit of premature optimisation there. I doubt we'd get anywhere close to the size of loops needed for it to be noticable (I have other concerns if someone has that many players installed!).

Side note, had never thought of using `and` like that for variable assignment before. That's really clever. Will need to remember that for the future.

I'm happy with everything here and will merge/release. I'm sure I'll hear quick enough if we've broken something but all seems to be working well with my tests. Really great additions and also very keen for these corresponding UpNext improvements.